### PR TITLE
Added Modular Exponential [C]

### LINKED ISSUE #208: Add Modular Exponentiation [C]
[http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/](http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/)

### PR DESCRIPTION
**Pull request by aashutoshrathi**
Fixes #208 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

File: modular_exponential/modular_exponential.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include <stdio.h>
+ 
+// Time Complexity : O(log (power)) 
+long long modularExponential(int base, long power, long long mod) {
+    long long answer = 1;    
+    base = base % mod;   
+    while (power) {
+        if (power & 1)
```
Comment (singhpratyush) [inline modular_exponential/modular_exponential.c]: Please enclose in a block.

Comment (aashutoshrathi) [inline modular_exponential/modular_exponential.c]: Do you mean I should add `{` in if block

Comment (singhpratyush) [inline modular_exponential/modular_exponential.c]: Yes.

Comment (aashutoshrathi) [inline modular_exponential/modular_exponential.c]: Done 👍

File: modular_exponential/modular_exponential.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
+ 
+// Time Complexity : O(log (power)) 
+long long modularExponential(int base, long power, long long mod) {
```
Comment (aviaryan) [inline modular_exponential/modular_exponential.c]: Make base as `long long`. 
Why?
Suppose base = 10^7 .. base * base overflows.

**Pull request merged**